org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/model/ContentModelManager.java

```diff
@@ -310,6 +310,10 @@ private CMDocument findCMDocument(String uri, String publicId, String systemId,
 				Path file = cacheResolverExtension.getCachedResource(resolvedUri);
 				if (file != null) {
 					cmDocument = modelProvider.createCMDocument(file.toUri().toString());
+				} else {
+					// It's not a file URL, and not a remote URL (ftp, http, https)? It's something else that might have a registered URL Handler.

```

Comment:
Why you need that? If you need it, please do that in an another PR (with test please).

Reply:
If schema is located inside a dependency JAR of a project the schema location URL is `project-aware://<project name>/<path>`. If you prefer a separate PR for this I could move it indeed

Reply:
Yes please it will be more easy to review this usecase and add test for that. Thanks.

Reply:
Here is the PR for this: https://github.com/eclipse/lemminx/pull/906